### ISSUE TITLE
Monitoring the progress of a simulation with SynchronousDynamics

### ISSUE THREAD

**Issue by ElanOcher** [simoninireland/epydemic#11]
Dear Simon,

I have started using your package recently and it is great. I had an issue with monitoring the progress of a simulation with SynchronousDynamics. Is it supported when running monitored simulation?
When I run my model with synchronous/stochastic dynamics it works in both ways. But when I try to run a monitored simulation the stochastic works and synchronous doesn't. 

This is a code snippet:
```
m = SIRQ()
p = epydemic.ProcessSequence([m, epydemic.Monitor()])

T = 1000000
num_of_points = 1000
deltaM = T / num_of_points
param[epydemic.Monitor.DELTA] = deltaM


e = epydemic.StochasticDynamics(p, g)
e.process().setMaximumTime(T)
rc = e.set(param).run()
```
If I change the StochasticDynamics command to e = epydemic.SynchronousDynamics(p, g) the model remains static and not infection occurs (I used printing to check in the event functions and they are not called).  What can I do to make it work?

**Comment by simoninireland** [comment]
Hi Elan,

Thanks for this -- I'll have a look and get back to you.

Cheers,

-- Simon

**Comment by simoninireland** [comment]
OK, found out what the problem was: something the tests should have picked up, obviously, so thanks very much for raising the issue. If you upgrade to the latest version (1.9.2) it should be working.

Most of the time we don't use the SynchronousDynamics class, as the StochasticDynamics approach is more statistically exact.

Thanks for your interest in epydemic: I'd love to hear more about what you're using it for, and please let me know if there are any more issues.

Cheers,

-- Simon

**Comment by simoninireland** [comment]
OK, so we solved one problem and opened-up another by the looks of it :-). I'll take a look. In the meantime I hope the StochasticDynamics is useful to you: although it's more complicated to understand it's definitely a better simulation approach.

Cheers,

-- Simon

**Comment by ElanOcher** [comment]
Yes, the StochasticDynamics is very useful for me! And I agree it is better simulation approach statistically. Thank you very much for looking into the issue.

**Comment by simoninireland** [comment]
Think that's now fixed (in version 1.9.4) -- I'd made a mistake between probabilities and rates. I've also added a note about the two different kinds of dynamics. If you could check it that'd be great -- I'll leave this issue open in case anything else turns up.

Cheers,

-- Simon

**Comment by ElanOcher** [comment]
Thank you Simon, seems to work great! Sorry it took me long time to check, I am barely with a computer this week.
SynchronousDynamics:
![image](https://user-images.githubusercontent.com/99040541/160808676-6fa949a0-e659-409b-b897-78b5f42baaa9.png)
StochasticDynamics :
![image](https://user-images.githubusercontent.com/99040541/160808402-63544ef3-f64c-433f-9bf6-c9c84c0b44f5.png)

Cheers,

--Elan
